source/extensions/filters/http/gcp_authn/gcp_authn_filter.cc

```diff
@@ -13,9 +13,14 @@ namespace Extensions {
 namespace HttpFilters {
 namespace GcpAuthn {
 namespace {
-void addTokenToRequest(Http::RequestHeaderMap& hdrs, absl::string_view token_str) {
-  std::string id_token = absl::StrCat("Bearer ", token_str);
-  hdrs.setCopy(authorizationHeaderKey(), id_token);
+void addTokenToRequest(Http::RequestHeaderMap& hdrs, absl::string_view token_str,
+                       absl::string_view header_name) {
+  if (!header_name.empty()) {
+    hdrs.setCopy(Http::LowerCaseString(header_name), token_str);
+  } else {
+    std::string id_token = absl::StrCat("Bearer ", token_str);
```

Comment:
It seems odd to me that we prefix with "Bearer" only in this case, but not in the other case. If there is a reason for that can you document that also?

Reply:
For our use case we would like to have it like this.
```
<custom header name>: <token>
```
But it could be that others may want the `Bearer` prefix when using a custom header name.
Wdyt about taking some inspiration from the [JWT filter](https://www.envoyproxy.io/docs/envoy/v1.24.0/api-v3/extensions/filters/http/jwt_authn/v3/config.proto#extensions-filters-http-jwt-authn-v3-jwtheader) and allow configuring the prefix for this filter as well?
So the default would something like:
```yaml
token_header:
  name: "Authorization"
  value_prefix: "Bearer "
```
and then for our use case we can just do this:
```yaml
token_header:
  name: "our-custom-header"
  value_prefix: ""
```

Reply:
Seems like a good compromise to me, I will go ahead and update the code.

Reply:
Wouldn't there also be some value in being consistent with the naming in `extensions.filters.http.jwt_authn.v3.JwtHeader` (which uses `value_prefix` for this)?
I could imagine cases where gcp_authn and jwt_authn are used together in a client/server pair and in that case some consistency might make it slightly more user friendly.

Reply:
I didn't check the `jwtHeader`. Yea, I agree it is good to be consistent with the naming there.

Reply:
For the sake of consistency, I left `value_prefix` alone :-)